### PR TITLE
coreboot-toolchain: 4.16 -> 4.18

### PR DESCRIPTION
**Pull request by jmbaur**
###### Description of changes

Updates the coreboot-toolchain with the provided
pkgs/development/tools/misc/coreboot-toolchain/update.sh.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/misc/coreboot-toolchain/default.nix
Comment (felixsinger) [inline pkgs/development/tools/misc/coreboot-toolchain/default.nix:51]: Keep `stable.nix` as filename.

File: pkgs/development/tools/misc/coreboot-toolchain/sources.nix
Comment (felixsinger) [inline pkgs/development/tools/misc/coreboot-toolchain/sources.nix:45]: Remove, not used.

File: pkgs/development/tools/misc/coreboot-toolchain/sources.nix
Comment (felixsinger) [inline pkgs/development/tools/misc/coreboot-toolchain/sources.nix:52]: Remove, not used.

File: pkgs/development/tools/misc/coreboot-toolchain/sources.nix
Comment (felixsinger) [inline pkgs/development/tools/misc/coreboot-toolchain/sources.nix:59]: Remove, not used.

File: pkgs/development/tools/misc/coreboot-toolchain/sources.nix
Comment (felixsinger) [inline pkgs/development/tools/misc/coreboot-toolchain/sources.nix:66]: Remove, not used.

File: pkgs/development/tools/misc/coreboot-toolchain/sources.nix
Comment (felixsinger) [inline pkgs/development/tools/misc/coreboot-toolchain/sources.nix:73]: Remove, not used.

File: pkgs/development/tools/misc/coreboot-toolchain/sources.nix
Comment (felixsinger) [inline pkgs/development/tools/misc/coreboot-toolchain/sources.nix:80]: Remove, not used.

**Review by felixsinger — CHANGES_REQUESTED** [review]

**Comment by felixsinger** [comment]
@GrahamcOfBorg build coreboot-toolchain.i386 coreboot-toolchain.arm coreboot-toolchain.aarch64 coreboot-toolchain.riscv coreboot-toolchain.ppc64 coreboot-toolchain.nds32le coreboot-toolchain.x64

**Comment by felixsinger** [comment]
@GrahamcOfBorg build coreboot-toolchain.i386 coreboot-toolchain.arm coreboot-toolchain.aarch64 coreboot-toolchain.riscv coreboot-toolchain.ppc64 coreboot-toolchain.nds32le coreboot-toolchain.x64

**Comment by felixsinger** [comment]
Let's wait until the following two patches got merged. The 202203 has a regression and shouldn't be used.

https://review.coreboot.org/c/coreboot/+/70856
https://review.coreboot.org/c/coreboot/+/70857

**Comment by jmbaur** [comment]
Closed in favor of #212814

**Pull request closed without merging**
